### PR TITLE
Rework introduction page on VEE Porting Guide

### PR DESCRIPTION
**Pull request by acolleux-microej**
# CHANGES

image

### CODE REVIEW

File: VEEPortingGuide/introduction.rst [hunk 1]
```diff
@@ -0,0 +1,187 @@
+============
+Introduction
+============
+
+A MicroEJ Architecture is a software package that includes the MicroEJ Runtime port to a specific target Instruction Set Architecture (ISA) and C compiler.
+It contains a set of libraries, tools and C header files. The MicroEJ Architectures are provided by MICROEJ SDK.
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:6]: MicroEJ Architectures are provided by MICROEJ SDK.

File: VEEPortingGuide/introduction.rst [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+A VEE Port is a MicroEJ Architecture port for a custom device.
+
+It relies on C drivers (aka low level LL drivers) for each VEE Port feature.
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:10]: a.k.a. Low Level (LL) drivers

File: VEEPortingGuide/introduction.rst [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+These drivers are implemented in the VEE Pot BSP project. This project is edited in the C compiler IDE/dev environment (e.g. KEIL, GCC, IAR).
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:11]: Port

File: VEEPortingGuide/introduction.rst [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+E.g. the MicroUI library LED feature will require a ``LLUI_LED.c`` that implements the native on/off IO drive.
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:12]: I don't think we should start a sentence with a contraction, use For Example.

File: VEEPortingGuide/introduction.rst [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+* a MicroEJ Architecture (MCU ISA and C compiler)
+* an optional RTOS (e.g. FreeRTOS - note: the MicroEJ Runtime can run bare metal)
+* a device: the OS bring up code that is device specific (e.g. the MCU specific code/IO/RAM/Clock/Middleware… configurations)
+
+MicroEJ Corp. provides MicroEJ Evaluation Architectures at https://repository.microej.com/modules/com/microej/architecture,
+and `VEE Port Examples`_ for various evaluation boards.
+
+The VEE Porting Guide explains how the core features are
+accessed, configured and used for creating a port of MICROEJ VEE (VEE Port) on your dedicated device.
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:24]: to create

File: VEEPortingGuide/introduction.rst [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+* a MicroEJ Architecture (MCU ISA and C compiler)
+* an optional RTOS (e.g. FreeRTOS - note: the MicroEJ Runtime can run bare metal)
+* a device: the OS bring up code that is device specific (e.g. the MCU specific code/IO/RAM/Clock/Middleware… configurations)
+
+MicroEJ Corp. provides MicroEJ Evaluation Architectures at https://repository.microej.com/modules/com/microej/architecture,
+and `VEE Port Examples`_ for various evaluation boards.
+
+The VEE Porting Guide explains how the core features are
+accessed, configured and used for creating a port of MICROEJ VEE (VEE Port) on your dedicated device. 
+It also explains how an Application can interoperate with native code, and
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:25]: interact with

File: VEEPortingGuide/introduction.rst [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+Semantics of implemented Foundation Libraries are described in
+their respective chapters as well as the required Abstraction Layers APIs for porting them 
+to different targets. 
+
+.. _VEE Port Examples: https://github.com/microej?q=vee&type=all&language=&sort=
+
+VEE Port Build Process
+======================
+
+The following figure shows the overall process to obtain an Executable file to deploy on a device.
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:38]: The following figure shows the overall process to create an Executable file to deploy on a device.

File: VEEPortingGuide/introduction.rst [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+
+5. Compile the BSP and link it with the Application object file that was
+   built previously in step 4 to produce an Executable.
+
+6. Final step: Deploy the Executable (i.e. the binary application) onto a device.
+
+Create a VEE Port for a Custom Device
+=====================================
+
+VEE Port creation can be either done from scratch or modifying an existing VEE Port example project.
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:70]: VEE Port creation can either be done from

File: VEEPortingGuide/introduction.rst [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+* RAM 8/16/32 -bit read/write operations (internal and external if any)
+* EEMBC Coremark benchmark to verify the CPU/buses/memory/compiler configuration
+
+A VEE Port Example is already available for the same MCU/RTOS/C Compiler
+------------------------------------------------------------------------
+
+This is the fastest way: `VEE Port Examples`_ are usually provided for a silicon vendor evaluation board.
+Import this VEE Port in MICROEJ SDK.
+
+As the MCU, RTOS and compiler are the same, only the device specific code needs to be changed (external RAM, external oscillator, communication interfaces).
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:90]: device-specific

File: VEEPortingGuide/introduction.rst [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+
+VEE Port Configuration
+::::::::::::::::::::::
+
+* Modify the ``xxx-configuration`` project to match the device features and its associated configuration (e.g. GUI, Networking, ...).
+
+Refer to MICROEJ SDK5 :ref:`platform_module_configuration` or MICROEJ SDK 6 VEE Porting Guide to learn more about VEE Port configuration.
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:97]: There is no MICROEJ SDK 6 VEE Porting Guide.

File: VEEPortingGuide/introduction.rst [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+  * edit the scatter file/link options
+  * edit the compilation options
+
+* create/review/change the platform Low Level C drivers. 
+  They must match the device components and the MCU IO pin assignment:
+  
+  .. note::
+    
+    A number of ``LL*.h`` files are referenced from the project. 
+    Implement the function prototypes declared there so that the JVM can delegate the relevant operations to the provided BSP C functions.
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:117]: Shouldn't we talk about MEJ32 instead of JVM?

File: VEEPortingGuide/introduction.rst [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
+At this point, consider either to modify the closest VEE Port:
+
+* In the SDK: modify the VEE Port configuration.
+* in the C IDE: start from an empty project that match with the MCU.
+
+Or to start from scratch a new VEE Port:
+
+* In the SDK: create the VEE Port and refer to the selected VEE Port as a model for implementation
+  (refer to MICROEJ SDK5 :ref:`platform_configuration_creation` or MICROEJ SDK6 VEE Porting Guide)
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:143]: SD6 VEE Porting Guide, where are you, brother?

File: VEEPortingGuide/introduction.rst [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+* in the C IDE: start from an empty project and implement the drivers of each of the LL drivers API. 
+  
+  Make sure to link with:
+
+  * the ``microejruntime.a`` that runs the JVM for the MCU Architecture
```
Comment (gharelMicroEJ) [inline VEEPortingGuide/introduction.rst:148]: JVM

**Review by gharelMicroEJ — APPROVED** [review]

File: VEEPortingGuide/introduction.rst (hunk 5 above)
Comment (acolleux-microej) [reply VEEPortingGuide/introduction.rst]: Fixed

**Pull request merged**